ui/src/components/FieldSearch.js

```diff
@@ -3,20 +3,16 @@ import Select from "react-select";
 
 class FieldSearch extends React.Component {
   constructor(props) {
-    const fields = props.fields;
     super(props);
-    this.handleChange = props.handleChange;
-    this.state = {
-      fields: fields
-    };
   }
 
   render() {
     return (
       <Select
         isMulti="true"
-        onChange={this.handleChange}
-        options={this.state.fields}
+        onChange={this.props.handleChange}
+        options={this.props.fields}
+        value={this.props.extraFacetsOptions}
```

Comment:
I actually don't find this intuitive. I think the FieldSearch component is the source of truth for storing the selectedOptions. The FieldSearch component passes the selectedOptions to the parent component in the event handler for onChange(). The parent component is storing the selectedOptions in it's state. 
But in this PR, the parent is setting the value of the select component, which seems like it's overwriting the source. IMO, this is also prone to bugs. If the parent component modifies the selectedOptions then the value of the select box will also change.

Reply:
Discussed offline. Yeah, this aspect of Reactive is unintuitive in the beginning. Intuitively, one would expect that FieldSearch would be source of truth for selected options, and the FacetsGrid would also be source of truth for selected options (since it has to display new facets). When user adds a new option, FieldSearch would notify FacetsGrid. In practice, having different components both maintain the same source of truth leads to bugs, because our code might not keep them in sync.

The React way is, react-select callback tells App that selection changed. App tells all children (including FieldSearch) about new options. It is more roundabout than FieldSearch just knowing itself which options are selected, but it leads to simpler code over time.